### ISSUE TITLE
core dumped

### ISSUE THREAD

**Issue by hartmark** [desbma/motd#4]
Hi again :)

I have another core dumped issue now, I still haven't learnt rust but I have bisected to get the broken commit atleast for you.

```
───────────────────────────────────── Load ─────────────────────────────────────
Load avg 1min: 1.63, 5 min: 1.56, 15 min: 2.36
Tasks: 1268

───────────────────────────────── Memory usage ─────────────────────────────────
MemTotal: 31.26 GB
MemFree:  19.13 GB (61.2%)
Dirty:     3.29 MB ( 0.0%)
Cached:    9.10 GB (29.1%)
Buffers:   7.75 MB ( 0.0%)
▕█Used███Cached 9.1GB (29.1%)██              Free 19.1GB (61.2%)               ▏
memory allocation of 9223372036854775783 bytes failed
[2]    27308 abort (core dumped)  motd
────────────────────────────────── Arch news ───────────────────────────────────
No news to display.

```

```
markus@bernard ~/code/motd (git)-[1.1.2~2|bisect] % git bisect good              
557ec5af769ac82c056e702e073c44dd3d9d5ef1 is the first bad commit
commit 557ec5af769ac82c056e702e073c44dd3d9d5ef1
Author: desbma <desbma@users.noreply.github.com>
Date:   Mon Jul 19 01:39:20 2021 +0200

    Fix mix of SI/IEC byte units

 Cargo.lock  | 143 +++++++++++++++++++++++++++++-------------------------------
 Cargo.toml  |   1 -
 src/fmt.rs  |  33 ++++++++++++++
 src/fs.rs   |  29 ++++++------
 src/main.rs |   1 +
 src/mem.rs  |  41 +++++------------
 src/net.rs  |  32 ++++----------
 7 files changed, 138 insertions(+), 142 deletions(-)
 create mode 100644 src/fmt.rs


```

**Comment by desbma** [comment]
Hi, can you run a debug build and post the full stacktrace?
```
git clone https://github.com/desbma/motd
cd ./motd
RUST_BACKTRACE=full cargo run
```

EDIT: I would also need the output of `cat /proc/meminfo`.

**Comment by hartmark** [comment]
Sorry, thought I attached the full stacktrace already.

```
% RUST_BACKTRACE=full cargo run
   Compiling motd v1.1.2 (/home/markus/code/motd)
    Finished dev [unoptimized + debuginfo] target(s) in 1.83s
     Running `target/debug/motd`
───────────────────────────────────── Load ─────────────────────────────────────
Load avg 1min: 2.12, 5 min: 1.27, 15 min: 0.54
Tasks: 1025

───────────────────────────────── Memory usage ─────────────────────────────────
MemTotal: 31.26 GB
MemFree:  26.40 GB (84.5%)
Dirty:    24.48 MB ( 0.1%)
Cached:    2.87 GB ( 9.2%)
Buffers:  10.47 MB ( 0.0%)
▕Used█Cached█                       Free 26.4GB (84.5%)                        ▏
thread 'main' panicked at 'attempt to subtract with overflow', src/fs.rs:131:29
stack backtrace:
   0:     0x563bcefb4620 - std::backtrace_rs::backtrace::libunwind::trace::hfc9639750d1d96c4
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x563bcefb4620 - std::backtrace_rs::backtrace::trace_unsynchronized::h5ec4a0e17ab2632f
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x563bcefb4620 - std::sys_common::backtrace::_print_fmt::h3a69f32931dd136f
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x563bcefb4620 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf88af4c84d6c5709
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x563bcefd9f2f - core::fmt::write::h4a5435262f6da981
                               at /build/rust/src/rustc-1.53.0-src/library/core/src/fmt/mod.rs:1094:17
   5:     0x563bcefaf6d5 - std::io::Write::write_fmt::h0e8e51e97f3f7842
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/io/mod.rs:1584:15
   6:     0x563bcefb681b - std::sys_common::backtrace::_print::heef60f04efc00f0e
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x563bcefb681b - std::sys_common::backtrace::print::hdea28e405156565e
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x563bcefb681b - std::panicking::default_hook::{{closure}}::h342b43211f5250e9
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:208:50
   9:     0x563bcefb62ed - std::panicking::default_hook::h794beb2216e46c37
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:225:9
  10:     0x563bcefb6e3d - std::panicking::rust_panic_with_hook::ha63044836e5bdf5f
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:591:17
  11:     0x563bcefb6997 - std::panicking::begin_panic_handler::{{closure}}::hc3b84897e913c1b6
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:495:13
  12:     0x563bcefb4adc - std::sys_common::backtrace::__rust_end_short_backtrace::h9f87358fdd9e1308
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x563bcefb6929 - rust_begin_unwind
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:493:5
  14:     0x563bcedb62c1 - core::panicking::panic_fmt::h1ff6e3ae0bf66df6
                               at /build/rust/src/rustc-1.53.0-src/library/core/src/panicking.rs:92:14
  15:     0x563bcedb620d - core::panicking::panic::h256a2315c84cd28e
                               at /build/rust/src/rustc-1.53.0-src/library/core/src/panicking.rs:50:5
  16:     0x563bcee14195 - motd::fs::get_fs_bar::h116816ddd9c82838
                               at /home/markus/code/motd/src/fs.rs:131:29
  17:     0x563bcee14fba - motd::fs::output_fs_info::hae4bfd30b7a0c6c7
                               at /home/markus/code/motd/src/fs.rs:199:13
  18:     0x563bcee20367 - motd::main::{{closure}}::hed0f257e004632f1
                               at /home/markus/code/motd/src/main.rs:416:45
  19:     0x563bcede1d6f - core::result::Result<T,E>::map::h33a850e5de3884bf
                               at /build/rust/src/rustc-1.53.0-src/library/core/src/result.rs:504:25
  20:     0x563bcee1e3d2 - motd::main::hf7ab3d1306029653
                               at /home/markus/code/motd/src/main.rs:416:29
  21:     0x563bcedcac9b - core::ops::function::FnOnce::call_once::h2c1976893cbc9fab
                               at /build/rust/src/rustc-1.53.0-src/library/core/src/ops/function.rs:227:5
  22:     0x563bcedc6b0e - std::sys_common::backtrace::__rust_begin_short_backtrace::h9b01c456e55c1383
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/sys_common/backtrace.rs:125:18
  23:     0x563bcedd4091 - std::rt::lang_start::{{closure}}::hfee57d43abbb37f4
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/rt.rs:49:18
  24:     0x563bcefb723a - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h2df0b8901ef38a06
                               at /build/rust/src/rustc-1.53.0-src/library/core/src/ops/function.rs:259:13
  25:     0x563bcefb723a - std::panicking::try::do_call::h2974db5c790e60de
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:379:40
  26:     0x563bcefb723a - std::panicking::try::ha03ebed38c5b8455
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panicking.rs:343:19
  27:     0x563bcefb723a - std::panic::catch_unwind::hb59b38016092aaf1
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/panic.rs:431:14
  28:     0x563bcefb723a - std::rt::lang_start_internal::h87f771367383f37c
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/rt.rs:34:21
  29:     0x563bcedd4070 - std::rt::lang_start::h8418ce8e1b4c7368
                               at /build/rust/src/rustc-1.53.0-src/library/std/src/rt.rs:48:5
  30:     0x563bcee2052c - main
  31:     0x7f8e75e08b25 - __libc_start_main
  32:     0x563bcedb6a3e - _start
  33:                0x0 - <unknown>
```


```
% cat /proc/meminfo
MemTotal:       32774996 kB
MemFree:        27591560 kB
MemAvailable:   30306112 kB
Buffers:           10720 kB
Cached:          3055048 kB
SwapCached:            0 kB
Active:          1107644 kB
Inactive:        3466388 kB
Active(anon):       1608 kB
Inactive(anon):  1550232 kB
Active(file):    1106036 kB
Inactive(file):  1916156 kB
Unevictable:         176 kB
Mlocked:             176 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:              1836 kB
Writeback:             0 kB
AnonPages:       1508576 kB
Mapped:           546104 kB
Shmem:             43580 kB
KReclaimable:     114636 kB
Slab:             308816 kB
SReclaimable:     114636 kB
SUnreclaim:       194180 kB
KernelStack:       17488 kB
PageTables:        22600 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:    16387496 kB
Committed_AS:    5977492 kB
VmallocTotal:   34359738367 kB
VmallocUsed:       81036 kB
VmallocChunk:          0 kB
Percpu:            27008 kB
HardwareCorrupted:     0 kB
AnonHugePages:         0 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
FileHugePages:         0 kB
FilePmdMapped:         0 kB
CmaTotal:              0 kB
CmaFree:               0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB
DirectMap4k:      869432 kB
DirectMap2M:     8470528 kB
DirectMap1G:    24117248 kB
```

**Comment by desbma** [comment]
This is fixed in v1.1.3, thanks for the report.
